File: cloud-manager/justice.k8s/src/main/java/com/lnjoying/justice/cluster/manager/common/ClusterStatus.java
package com.lnjoying.justice.cluster.manager.common;

import java.util.HashMap;
import java.util.Map;

public enum  ClusterStatus
{
    CREATED(               0, "实例已创建", "Created"),
    SPAWN(                 1, "发起调度", "Spawn"),
    ASSIGNING(             2, "分配节点中", "Assigning"),
    ASSIGNED(              3, "已分配节点", "Assigned"),
    PLANNING(              4, "构建任务中", "building deploy plan"),
    PLANNED(               5, "构建任务完成", "build deploy plan success"),
    PLAN_FAILED(           6, "构建任务失败", "build deploy plan failed"),
    DEPLOYING(             7, "部署中", "deploying"),
    DEPLOYED(              8, "部署完成", "deployed"),
    DEPLOY_PARTIAL_SUCCESS(9, "部分部署成功", "deploy partial success"),
    IMPORTED(             10, "导入完成", "imported"),
    DEPLOY_FAILED(        11, "部署失败", "deploy failed"),
    ASSIGNED_FAILED(      12, "分配节点失败", "Assigned failed"),
    BUILD_FAILED(         13, "构建失败", "build failed"),
    
    REMOVING(             90, "删除中", "Removing"),
    REMOVE_FAILED(        91, "删除失败", "Remove failed"),
    REMOVED(              99, "已删除", "Removed");
    
    ClusterStatus(int code, String cnDesc, String enDesc)
    {
        this.setCode(code);
        this.message = new HashMap<String, String>()
        {
            {
                put("cn", cnDesc);
                put("en", enDesc);
            }
        };
    }
    
    public static ClusterStatus fromCode(int code)
    {
        for (ClusterStatus current : ClusterStatus.values())
        {
            if (current.getCode() == code)
            {
                return current;
            }
        }
        throw new RuntimeException("Invalid error code: " + code);
    }
    
    public void setCode(int code)
    {
        this.code = code;
    }
    
    public int getCode()
    {
        return code;
    }
    
    public Map<String, String> getMessage()
    {
        return message;
    }

    /** Numeric code */
    private int code;
    
    /** Error message */
    private Map<String, String> message;
}

File: web/src/utils/status/clusterStatus.js
import i18n from "@/utils/i18n/index.js";

const codeMessageMap = {
  0: i18n.t("clusterStatus.Created"),
  1: i18n.t("clusterStatus.Spawn"),
  2: i18n.t("clusterStatus.Assigning"),
  3: i18n.t("clusterStatus.Assigned"),
  4: i18n.t("clusterStatus.buildingDeployPlan"),
  5: i18n.t("clusterStatus.buildDeployPlanSuccess"),
  6: i18n.t("clusterStatus.buildDeployPlanFailed"),
  7: i18n.t("clusterStatus.deploying"),
  8: i18n.t("clusterStatus.deployed"),
  9: i18n.t("clusterStatus.deployPartialSuccess"),
  10: i18n.t("clusterStatus.imported"),
  11: i18n.t("clusterStatus.deployFailed"),
  12: i18n.t("clusterStatus.AssignedFailed"),
  13: i18n.t("clusterStatus.buildFailed"),
  90: i18n.t("clusterStatus.Removing"),
  91: i18n.t("clusterStatus.RemoveFailed"),
  99: i18n.t("clusterStatus.Removed"),
};
const codeMessageTagMap = {
  0: "success",
  1: "",
  2: "",
  3: "success",
  4: "",
  5: "success",
  6: "danger",
  7: "",
  8: "success",
  9: "warning",
  10: "success",
  11: "danger",
  12: "danger",
  13: "danger",
  90: "warning",
  91: "danger",
  99: "",
};
const showClusterStatusMessage = (code, type) => {
  if (type == "tag") {
    return codeMessageTagMap[JSON.stringify(code)] != undefined
      ? codeMessageTagMap[JSON.stringify(code)]
      : "info";
  } else {
    return (
      codeMessageMap[JSON.stringify(code)] ||
      i18n.t("statusAndType.unknownStatus")
    );
  }
};

export default showClusterStatusMessage;
